### PR TITLE
Fix #2363 per theme default avatar

### LINKED ISSUE #2363: Per theme default avatar
Make it possible to use something like `{theme}` for the default avatar as suggested here: http://community.mybb.com/thread-172906.html

### PR DESCRIPTION
**Pull request by Destroy666x**
I didn't include `$change_dir` because it doesn't seem to be implemented
correctly anyways (I'll create an issue later).

https://github.com/mybb/mybb/issues/2363

### CODE REVIEW

**Comment by Cu8eR** [comment]
Looks fine

**Comment by Destroy666x** [comment]
The PR is finished after the last commit

**Comment by JordanMussi** [comment]
@Destroy666x, I suggest using `<theme>` instead of `{theme}` since `<lang:foobar>` is [already used](https://github.com/mybb/mybb/blob/mybb_1807/inc/class_language.php) for replacing language variables in various things. Also using curly braces may cause confusion with `{$theme}`.

Another thing, I think the following setting description will make it completely clear what happens (you also have `use use` in your description - I hate typos too! :stuck_out_tongue:):
`If the user does not set a custom avatar this image will be used instead. To have a different default avatar for each theme, you may use <strong>&lt;theme&gt;</strong> which is replaced with the directory of the active theme.`

Which would display as:
> If the user does not set a custom avatar this image will be used instead. To have a different default avatar for each theme, you may use **&lt;theme&gt;** which is replaced with the directory of the active theme.

File: admin/modules/user/users.php [hunk 1]
```diff
@@ -3587,7 +3587,7 @@ function build_users_view($view)
 			}
 			if(!$user['avatar'])
 			{
-				$user['avatar'] = "../".$mybb->settings['useravatar'];
+				$user['avatar'] = "../".str_replace('{theme}', 'images', $mybb->settings['useravatar']);
```
Comment (JordanMussi) [inline admin/modules/user/users.php]: Have you created an issue for not using `format_avatar()` here (and the one above) or could this be fixed with #2363?

**Comment by Destroy666x** [comment]
> @Destroy666x, I suggest using <theme> instead of {theme} since <lang:foobar> is already used for replacing language variables in various things. Also using curly braces may cause confusion with {$theme}.

I don't know why `<theme>` would be better than `{theme}` which is used literally everywhere for replacement.. Consistency is important as you say and you can easily check that this is not the only place with `{theme}`, I even mentioned placed where it's used in the PR. And for language folders `{lang}` is used, I haven't seen the one you linked presented to the user anywhere. 

> you also have use use in your description - I hate typos too!

Thanks, corrected. The rest of the description is copied from another place that uses `{theme}` and IMO it's understandable.

File: admin/modules/user/users.php (hunk 1 above)
Comment (Destroy666x) [inline admin/modules/user/users.php]: It's not that important, but I created an issue: https://github.com/mybb/mybb/issues/2367

**Comment by JordanMussi** [comment]
Okies, I haven't been around for a while! Didn't remember `{theme}` was already used elsewhere...

**Comment by dequeues** [comment]
This has conflicts which need to be resolved.

**Comment by Destroy666x** [comment]
Well, conflict is fixed but rebase for some reason messed everything up, so I'll recreate it

**Pull request closed without merging**